libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.kt

```diff
@@ -209,13 +210,19 @@ open class SalesforceSDKManager protected constructor(
      */
     private var showDeveloperSupportBroadcastIntentReceiver: BroadcastReceiver? = null
 
-    val webviewLoginActivityClass: Class<out Activity> = loginActivity ?: LoginActivity::class.java
+    val webViewLoginActivityClass: Class<out Activity> = loginActivity ?: LoginActivity::class.java
 
     /**
      * The class of the activity used to perform the login process and create
      * the account.
      */
-    val loginActivityClass: Class<out Activity> = nativeLoginActivity ?: webviewLoginActivityClass
+    val loginActivityClass: Class<out Activity> = nativeLoginActivity ?: webViewLoginActivityClass
+
+    /**
+     * ViewModel Factory the SDK will use in LoginActivity and composable functions.  Setting this will allow for
+     * visual customization without overriding LoginActivity.
+     */
+    var loginViewModelFactory = LoginViewModel.Factory
```

Comment:
A note on this: This can _optionally_ be set but I use this everywhere because the system is subtly different than what you might expect.  The OS ensures there is only ever one instance of a view model **but inheritance counts**! 

For example:  If I create a subclass of `LoginViewModel` and set it to the viewModel val in my `LoginActivity` subclass our base `LoginViewModel` will never be instantiated.  Explicitly calling `LoginViewModel.Factory` will return the instance of the subclass of our view model.  So Apps overriding `LoginActivity` don't need to set this value. 

What the OS does is pretty elegant but not at all obvious.